### DISCUSSION TITLE
if condition in getRelations()

### DISCUSSION THREAD

**Discussion by dharen008** [filamentphp/filament#12681] [Help]
### Package

Panel builder

### Package Version

3

### How can we help you?

is there a way to do this? i cant call the ownerRecord

public static function getRelations(): array
	{
		if ( $this->title == 'Employee' ) {
			return [
				RelationManagers\ManageByRelationManager::make(),
			];
		} else {
			return [
				RelationManagers\ManagesRelationManager::make(),
				RelationManagers\ManageByRelationManager::make(),
			];
		}
	}

**Comment by borex** [comment]
Add both relation managers to the getRelations function and in the ManagesRelationManager class add the following function:
`
    public static function canViewForRecord(Model $ownerRecord, string $pageClass): bool
    {
        if ($ownerRecord->title != 'Employee') return false;
        return parent::canViewForRecord($ownerRecord, $pageClass);
    }
`
This worked for me. I don't know it there is a better way of showing/hiding relation managers based on owner record condition.

**Comment by dharen008** [comment]
nice! thank you for the help! its on the documentation too
https://filamentphp.com/docs/3.x/panels/resources/relation-managers#conditionally-showing-relation-managers
